modules/aws/src/smithy4s/query/FormData.scala

```diff
@@ -0,0 +1,57 @@
+package smithy4s
+package aws.query
+
+import smithy4s.PayloadPath
+import smithy4s.http.internals.URIEncoderDecoder
+
+private[aws] sealed trait FormData extends Product with Serializable {
+  def render: String
+  def prepend(key: String): FormData
+  def prepend(index: Int): FormData
+}
+private[aws] object FormData {
+  final case object Empty extends FormData {
+    override def render: String = ""
+
+    override def prepend(key: String): FormData = this
+
+    override def prepend(index: Int): FormData = this
+  }
+
+  final case class SimpleValue(str: String) extends FormData {
+    override def render: String = URIEncoderDecoder.encode(str)
+
+    override def prepend(key: String): FormData = {
+      PathedValue(PayloadPath.fromString(key), str)
+    }
+
+    override def prepend(index: Int): FormData =
+      PathedValue(PayloadPath.fromString(index.toString), str)
+  }
+  final case class PathedValue(path: PayloadPath, value: String)
+      extends FormData {
+
+    /**
+     * @todo Understand the root reason and have a better solution for a workaround removing the '.' prefix.
+     */
+    override def render: String =
+      URIEncoderDecoder.encode(path.toString.stripPrefix(".")) + "=" +
```

Comment:
the `.` exist because this Path construct is used in json decoding and I wanted to make it look more like `jq`. Really you shouldn't rely on `.toString` there, but rather deconstruct the path and fold the segments, which would prevent instantiating a string only to strip it of one char.

Reply:
Done, perhaps not ideal.